include/deal.II/trilinos/nox.h

```diff
@@ -153,24 +176,24 @@ namespace TrilinosWrappers
     solve(VectorType &solution);
 
     /**
-     * A user function that computes the residual @p f based on the
-     * current solution @p x.
+     * A function object that users should supply and that is intended to
+     * compute the residual `dst = F(src)`.
      *
      * @note This function should return 0 in the case of success.
      */
-    std::function<int(const VectorType &x, VectorType &f)> residual;
+    std::function<int(const VectorType &src, VectorType &dst)> residual;
```

Comment:
The edited text makes sense, but I would like to think about the naming of variables. The documentation of the KINSOL class uniformly uses `F(u)`, so it would make sense to replace `x` -> `u` (and `f` -> `F`, if one really wanted). But I would prefer not to use `src` and `dst`.

I acknowledge that the KINSOL class is not entirely consistent in this either, and would gladly accept patches to this end.